### PR TITLE
refine ept_flush_leaf_page

### PR DESCRIPTION
**Pull request by lifeix**
Refine the logic how to skip the pSRAM region when flushing cache.

Tracked-On: #5330
Signed-off-by: Li Fei1 <fei1.li@intel.com>

### CODE REVIEW

File: hypervisor/arch/x86/guest/ept.c [hunk 1]
```diff
@@ -203,56 +203,31 @@ void ept_del_mr(struct acrn_vm *vm, uint64_t *pml4_page, uint64_t gpa, uint64_t
...
-		} else if (flush_base_hpa < sw_sram_top) {
-			if (flush_end_hpa <= sw_sram_top) {
-				flush_size = 0UL;
-			} else {
-				/* Only flush [sw_sram_top, flush_end_hpa) and ignore [flush_base_hpa, sw_sram_top) */
-				flush_base_hpa = sw_sram_top;
-				flush_size = flush_end_hpa - sw_sram_top;
-			}
+		if (base_hpa < sw_sram_bottom) {
+			/* Flush [base_hpa, min(end_hpa, sw_sram_bottom) */
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/ept.c:219]: The original comments are more readable than this.  Please use the style of original comments, and explain the case of situation, i.e. how the regions overlap or not etc...

Comment (lifeix) [reply hypervisor/arch/x86/guest/ept.c:219]: The logic in this patch doesn't  need to care whether the region is overlap with pSRAM or not. What it does is:
if there is one part in the region is  on left side of pSRAM region, flush it first; if there's one part in the region is on right side of pSRAM region, flush it then.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br>No New Function Declaration/Definition Mismatch

**Comment by wenlingz** [comment]
@dongyaozu please help approve this PR, thanks

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br>No New Function Declaration/Definition Mismatch

**Comment by wenlingz** [comment]
@dongyaozu please help approve this PR, thanks

**Comment by wenlingz** [comment]
@dongyaozu please help approve this PR, thanks

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br>No New Function Declaration/Definition Mismatch

**Review by dongyaozu — APPROVED** [review]

**Pull request merged**